R/revolver_DETindex.R

```diff
@@ -0,0 +1,82 @@
+#' helper function to calculate diversity indices
+#' 
+#' @param   x           revolver output 
+#' @param   n.boot      number of bootstraps (default is 1)
+#' @param   table       return a table of results? (default is FALSE) 
+#' @param   index       'Shannon' or 'VA' (default is Shannon) 
+#' @param   type        'before.expansion' or 'after.expansion' (default after)
+#' @param   driver.only use driver events only? (default is FALSE)
+#' 
+#' @return
+#' 
+#' @import  vegan
+#' @import  qualvar
+revolver_DETindex <- function(x, 
+                              n.boot=1, 
+                              table=FALSE, 
+                              index=c('Shannon','VA'), 
+                              type=c('after.expansion','before.expansion'),
+                              driver.only=FALSE) {
+    
+  type <- match.arg(type)
+  index <- match.arg(index)
+  feat <- revolver.featureMatrix(x)
+  how <- paste0(ifelse(driver.only, "driver", "all"), 
+                ifelse(type=="before.expansion", "before", "after"))
+
+  switch(how, 
```

Comment:
Did you only expanded the functions or you also changed something else?

Reply:
the DETindex function I just refactored so that I could separate it out from dumpStatistics. In the process of doing so, it made sense to break out some of the branches into helper functions; the logic remains the same as it was beforehand, it's just more explicitly encapsulated. 

In addition: 

rev_cohort/revolver_cohort alias removed per review
Misc column included in dumpStatistics as before per review
a couple of WARNINGs from R CMD check addressed (during testing)

Should be good to go.